presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaRecordSet.java

```diff
@@ -203,6 +203,9 @@ private boolean nextRow(MessageAndOffset messageAndOffset)
                 if (columnHandle.isInternal()) {
                     KafkaInternalFieldDescription fieldDescription = KafkaInternalFieldDescription.forColumnName(columnHandle.getName());
                     switch (fieldDescription) {
+                        case TIMESTAMP_FIELD:
+                            currentRowValuesMap.put(columnHandle, longValueProvider(messageAndOffset.message().timestamp()));
```

Comment:
What does `messageAndOffset.message().timestamp()` return? Is it in UTC or some other time zone? Maybe there is no timezone?

Do you need to handle `session.isLegacyTimestamp()`?

Reply:
- messageAndOffset.message().timestamp() returns epoch time. 
- As it was to get message timestamp from kafka so I didn't think of session related time, could you please let me know if you see the need of ession.isLegacyTimestamp() and if I am missing something here.

Reply:
If you use `TIMESTAMP_WITH_TIME_ZONE` data type then you don't need to use `session.isLegacyTimestamp()`. To generate presto long representation of `TIMESTAMP_WITH_TIME_ZONE` you can use `io.prestosql.spi.type.DateTimeEncoding#packDateTimeWithZone(long, java.lang.String)`.